### DISCUSSION TITLE
New to Valhalla - GTFS

### DISCUSSION THREAD

**Discussion by Sunraj751** [valhalla/valhalla#3875] [Q&A]
I came across Valhalla 2 days ago

1. I am currently working on something that involves **Public transit**, and i know that Valhalla uses Transitland as a provider for transit routes. What i want to know is if somehow i can use GTFS data instead of Transitland? I can't use Transitland because it doesn't have routes for all public transit companies. 
2. I am trying to understand the code, but looking at so many files is really overwhelming. Can someone suggest some way to understand the code, so that i am able to collaborate for some feature? My current approach is to read through documentation whilst making sense of the code. Any suggestions are welcome

Thanks!

**Comment by kevinkreiser** [comment]
@Sunraj751 i am working to merge a bunch of work to support gtfs directly that @chris-jpark started this past summer. the good news is its quite far along, the bad news is i cant tell you how soon it will merge. right now im debugging the data side and after that we'll need to likely debug the algorithms side. it could finish up quite quickly but could also drag out. its hard to say have to debug more to tell. other good news is that this is my main priority right now so its being shepherded. you can follow along over at #3700
